docs/DeveloperGuide.md

```diff
@@ -1,34 +1,956 @@
-# Developer Guide
+# Developer Guide for Zer0Note
 
-## Design & implementation
+[1. Introduction](#1-introduction) <br>
+&nbsp;&nbsp;[1.1. Welcome!](#11-welcome)<br>
+&nbsp;&nbsp;[1.2. How to use this document](#12-how-to-use-this-document)<br>
+[2. Setting up](#2-setting-up) <br>
+&nbsp;&nbsp;[2.1. Prerequisites](#21-prerequisites) <br>
+&nbsp;&nbsp;[2.2. Setting up the project in your computer](#22-setting-up-the-project-in-your-computer) <br>
+&nbsp;&nbsp;[2.3. Verifying the setup](#23-verifying-the-setup) <br>
+&nbsp;&nbsp;[2.4. Configure coding style](#24-configure-the-coding-style) <br>
+[3. Design](#3-design) <br>
+&nbsp;&nbsp;[3.1. Architecture](#31-architecture) <br>
+&nbsp;&nbsp;[3.2. Commands](#32-commands) <br>
+&nbsp;&nbsp;[3.3. Notebooks](#33-notebooks) <br>
+&nbsp;&nbsp;[3.4. Tasks](#34-tasks) <br>
+[4. Implementation](#4-implementation) <br>
+&nbsp;&nbsp;[4.1. Mode Switch Feature](#41-mode-switch-feature) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[4.1.1. Implementation](#411-implementation) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[4.1.2. Design Considerations](#412-design-considerations) <br>
+&nbsp;&nbsp;[4.2. Timetable Mode](#42-timetable-mode) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[4.2.1. Tasklist Management Feature](#421-tasklist-management-feature) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[4.2.1.1. Implementation](#4211-implementation) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[4.2.1.2. Design Considerations](#4212-design-considerations) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[4.2.2. Tag Feature](#422-tag-feature) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[4.2.2.1. Implementation](#4211-implementation) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[4.2.2.2. Design Considerations](#4212-design-considerations) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[4.2.2. List Feature](#421-tasklist-management-feature) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[4.2.2.1. Implementation](#4211-implementation) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[4.2.2.2. Design Considerations](#4212-design-considerations) <br>
+&nbsp;&nbsp;[4.3. Notebook Mode](#43-notebook-mode) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[4.3.1. Notebook Management Feature](#431-notebook-management-feature) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[4.3.1.1. Implementation](#4311-implementation)
+&nbsp;&nbsp;&nbsp;&nbsp;[4.3.2. Tag Feature](#432-tag-feature) <br>
```

Comment:
Is this line displayed the way you want it to be? Not sure if its unintentional but maybe it would look better on a line of its own?

Reply:
Looks like we forgot a `<br>` tag. Thanks for pointing this out.